linera-core/src/client.rs

```diff
@@ -573,6 +598,9 @@ where
                     continue; // These applications are already registered; skip register message.
                 }
             }
+            if !self.options.message_policy.accepts(&message) {
+                continue;
+            }
```

Comment:
So if the blanket policy is "ignore", we never get this far, and ignore everything, no matter what's in the allow list.
If it's "reject", we'll _reject_ tracked messages from the allow list and _ignore_ all others.

I imagine it would be useful to _accept_ all messages from the allow list and _reject_ (not ignore) all other tracked messages.

Reply:
And if it's "accept" we accept messages only from the allow list and ignore all others.

So I guess we need a different mode, like "restricted" which accepts the allow list and rejects everything else?

Perhaps a subsequent PR?